slapp/transforms/transform_pipeline.py

```diff
@@ -179,30 +195,114 @@ def set_webm_parallelization(self, data, **kwargs):
         return data
 
 
+def xform_from_slapp_db(db_conn: query_utils.DbConnection,
+                        segmentation_run_id: int) -> Tuple[List[ROI], Path]:
+
+    # get all ROI ids from this segmentation run
+    query_string = ("SELECT id FROM rois WHERE segmentation_run_id="
+                    f"{segmentation_run_id}")
+    entries = db_conn.query(query_string)
+    roi_ids = [i['id'] for i in entries]
+
+    rois = [ROI.roi_from_query(roi_id, db_conn) for roi_id in roi_ids]
+
+    # load and downsample the source video
+    query_string = ("SELECT * FROM segmentation_runs "
+                    f"WHERE id={segmentation_run_id}")
+    seg_query = db_conn.query(query_string)[0]
+
+    return (rois, Path(seg_query['source_video_path']))
+
+
+class ProdSegmentationRunManifestSchema(mm.Schema):
+    experiment_id = mm.fields.Int(required=True)
+    binarized_rois_path = argschema.fields.InputFile(required=True)
+    # TODO: Consider using H5InputFile field from ophys_etl_pipelines
+    traces_h5_path = mm.fields.Str(required=True)
+    # TODO: Consider using H5InputFile field from ophys_etl_pipelines
+    movie_path = argschema.fields.Str(required=True)
+    local_to_global_roi_id_map = mm.fields.Dict(required=True,
+                                                keys=mm.fields.Int(),
+                                                values=mm.fields.Int())
+
+    @mm.post_load
+    def load_rois(self, data, **kwargs) -> dict:
+        with open(data['binarized_rois_path'], 'r') as f:
+            data['binarized_rois'] = json.load(f)
+        return data
+
+    @mm.post_load
+    def get_movie_frame_shape(self, data, **kwargs) -> dict:
+        with h5py.File(data['movie_path'], 'r') as h5f:
+            data['movie_frame_shape'] = h5f['data'].shape[1:]
+        return data
+
+
+def xform_from_prod_manifest(prod_manifest_path: str
+                             ) -> Tuple[List[ROI], Path]:
+    with open(prod_manifest_path, 'r') as f:
+        prod_manifest = json.load(f)
+    prod_manifest = ProdSegmentationRunManifestSchema().load(prod_manifest)
+    id_map = prod_manifest['local_to_global_roi_id_map']
+
+    rois = []
+    for roi in prod_manifest['binarized_rois']:
+        if roi['id'] not in id_map:
+            # only make manifests for listed ROIs
+            continue
+        # ROI mask_matrix is in 'postage stamp' form so row and col
```

Comment:
Oh, I didn't realize that the sparse matrix didn't encode the full 512x512

Reply:
LIMS-style format has the `mask_matrix` which is cropped and the offset of that in the full frames is in the `x` and `y` fields. Here, Nick is deferring the full-form offsets down a few lines:
```
        # TODO: check if offset applied may be off by 1
        converted_roi = ROI(coo_rows=roi_stamp.row + roi['y'],
                            coo_cols=roi_stamp.col + roi['x']
```

Reply:
So, it is encoding the full frame